data_steward/cdr_cleaner/cleaning_rules/dedup_measurement_value_as_concept_id.py

```diff
@@ -0,0 +1,168 @@
+ON
+  m.value_as_concept_id = lm.value_as_concept_id
+)
+""")
+
+
+class DedupMeasurementValueAsConceptId(BaseCleaningRule):
+
+    def __init__(self,
+                 project_id,
+                 dataset_id,
+                 sandbox_dataset_id=None,
+                 table_namer=None):
+        """
+        Initialize the class with proper information.
+
+        Set the issue numbers, description and affected datasets. As other tickets may affect
+        this SQL, append them to the list of Jira Issues.
+        DO NOT REMOVE ORIGINAL JIRA ISSUE NUMBERS!
+        """
+        desc = ('Corrects VALUE AS CONCEPT ID to be the ANSWER concept ID')
+
+        super().__init__(issue_numbers=JIRA_ISSUE_NUMBERS,
+                         description=desc,
+                         affected_datasets=[cdr_consts.COMBINED],
+                         affected_tables=[MEASUREMENT],
+                         project_id=project_id,
+                         dataset_id=dataset_id,
+                         sandbox_dataset_id=sandbox_dataset_id,
+                         table_namer=table_namer)
+
+    def get_query_specs(self):
+        """
+        Return a list of dictionary query specifications.
+
+        :return:  A list of dictionaries. Each dictionary contains a single query
+            and a specification for how to execute that query. The specifications
+            are optional but the query is required.
+        """
+        queries_list = []
+
+        sandbox_query, update_query = dict(), dict()
+
+        sandbox_query[cdr_consts.QUERY] = SANDBOX_QUERY.render(
+            sandbox_dataset=self.sandbox_dataset_id,
+            sandbox_table=self.sandbox_table_for(MEASUREMENT),
+            project=self.project_id,
+            dataset=self.dataset_id,
+            pipeline_tables=PIPELINE_TABLES,
+            identical_labs_table=IDENTICAL_LABS_LOOKUP_TABLE)
+        queries_list.append(sandbox_query)
+
+        update_query[cdr_consts.QUERY] = UPDATE_QUERY.render(
+            project=self.project_id,
+            dataset=self.dataset_id,
+            sandbox_dataset=self.sandbox_dataset_id,
+            identical_labs_table=IDENTICAL_LABS_LOOKUP_TABLE)
+        queries_list.append(update_query)
+
+        return queries_list
+
+    def setup_rule(self, client, *args, **keyword_args):
+        try:
+            client.get_table(
+                f'{self.project_id}.{self.sandbox_dataset_id}.{IDENTICAL_LABS_LOOKUP_TABLE}'
+            )
+        except NotFound:
+            job = client.copy_table(
+                f'{self.project_id}.{PIPELINE_TABLES}.{IDENTICAL_LABS_LOOKUP_TABLE}',
+                f'{self.project_id}.{self.sandbox_dataset_id}.{IDENTICAL_LABS_LOOKUP_TABLE}'
+            )
+            job.result()
+            LOGGER.info(
+                f'Copied {PIPELINE_TABLES}.{IDENTICAL_LABS_LOOKUP_TABLE} to '
+                f'{self.sandbox_dataset_id}.{IDENTICAL_LABS_LOOKUP_TABLE}')
```

Comment:
Is there a reason not to do a create/replace of the lookup table here?
```python
client.copy_table(
                f'{self.project_id}.{PIPELINE_TABLES}.{IDENTICAL_LABS_LOOKUP_TABLE}',
                f'{self.project_id}.{self.sandbox_dataset_id}.{IDENTICAL_LABS_LOOKUP_TABLE}',
                job_config=CopyJobConfig(write_disposition=google.cloud.bigquery.WriteDisposition.WRITE_TRUNCATE)
            )
```
It seems strange that the primary scenario (i.e. table not loaded yet) throws/catches exception. Also if table does exist at destination I think it is more likely that we will want to overwrite it e.g. if the pipeline table is more up to date. Let me know what you think.

Reply:
Thanks for the suggestion Mark.